### ISSUE TITLE
Error with Unicorn: syslog already open

### ISSUE THREAD

**Issue by workgena** [rocketjob/rails_semantic_logger#52]
### Environment
production unicorn

gems:
* ruby 2.3.1p112
* rails 4.2.9
* semantic_logger 4.2
* rails_semantic_logger 4.1.3
* syslog_protocol 0.9.2
* unicorn (4.8.3)

```ruby
# config/environments/production.rb

  config.rails_semantic_logger.add_file_appender = false
  config.semantic_logger.add_appender(
    appender: SemanticLogger::Appender::Syslog.new(
      application: 'MYAPP-admin',
      facility: ::Syslog::LOG_LOCAL7
    )
  )
```

```ruby
# config/unicorn.conf.rb
worker_processes 4 # this is important
...
after_fork do |server, worker|
  # Re-open appenders after forking the process
  SemanticLogger.reopen
end
```

$ bundle exec unicorn -E production -c config/unicorn.rb

When running the unicorn server in several processes the logger does not work.
If add `SemanticLogger.reopen` i see error in `log/unicorn.stderr.log`

```
syslog already open (RuntimeError)
```

Similar to this issue: https://github.com/rocketjob/rails_semantic_logger/issues/26#issuecomment-343555763

**Comment by seanthingee** [comment]
I've also come across an issue like this using Passenger. I can also replicate the issue by loading up the rails console and adding the :syslog appender twice (or once, if the environment also adds :syslog appender).

- Passenger 5.2.3
- Ruby 2.4.1
- semantic_logger 4.2.0
- rails_semantic_logger 4.1.3
- syslog_protocol 0.9.2

I poked around and tracked this down to `semantic_logger`. I made a fork of semantic_logger and added a fix that calls `reopen` instead of `open` if `::Syslog` is already opened. It appears to have fixed it for me, but I don't know if this is really the right way to do it. Since `reopen` actually closes and opens syslog, it might be better to do nothing if it's found to be already opened.

https://github.com/thingeecorp/semantic_logger/commit/510895aa2e44e783e2127319ac2b0469eb67649c
